indico/modules/attachments/controllers/management/event.py

```diff
@@ -97,3 +104,20 @@ class RHPackageEventAttachmentsManagement(AttachmentPackageMixin, RHManageEventB
     wp = WPPackageEventAttachmentsManagement
     management = True
     ALLOW_LOCKED = True
+
+
+class RHSetUploadPermissions(RHManageEventBase):
+    def _process(self):
+        current_settings = attachments_settings.get_all(self.event)
+        form = SetUploadPermissionsForm(obj=FormDefaults(**current_settings))
+        if form.validate_on_submit():
+            attachments_settings.set_multi(self.event, form.data)
+            new_settings = attachments_settings.get_all(self.event)
+            flash(_('Upload permissions settings changed successfully'), 'success')
+            changes = {k: (v, new_settings[k]) for k, v in current_settings.items() if v != new_settings[k]}
+            if changes:
+                log_fields = {'managers_only': 'Only managers can upload attachments'}
+                self.event.log(EventLogRealm.management, LogKind.change, 'Attachments', 'Settings updated',
+                               session.user, data={'Changes': make_diff_log(changes, log_fields)})
+            return jsonify_data()
+        return jsonify_form(form)
```

Comment:
This is now the third place where we have this almost identical code; possibly more if we count places where we hardcoded single/few individual settings. Let's add a mixin (maybe in events/controllers/base.py) like this and make use of it in those places:

```python
class EditEventSettingsMixin:
    settings_proxy = None
    form_cls = None
    success_message = None
    log_module = None
    log_message = None
    log_fields = None
    
    def _process(self):
        current_settings = self.settings_proxy.get_all(self.event)
        form = self.form_cls(obj=FormDefaults(**current_settings))
        if form.validate_on_submit():
            self.settings_proxy.set_multi(self.event, form.data)
            new_settings = self.settings_proxy.get_all(self.event)
            flash(self.success_message, 'success')
            changes = {k: (v, new_settings[k]) for k, v in current_settings.items() if v != new_settings[k]}
            if changes:
                self.event.log(EventLogRealm.management, LogKind.change, self.log_module, self.log_message, session.user,
                               data={'Changes': make_diff_log(changes, self.log_fields)})
            return jsonify_data()
        return jsonify_form(form)
```

Note: This code is untested.